backend/src/test/java/com/example/cms/user/UserControllerTest.java

```diff
@@ -0,0 +1,541 @@
+                            "TEST_MODERATOR"+Instant.now().toString(),
+                            null,
+                            null,
+                            null,
+                            null,
+                            null,
+                            null,
+                            Role.MODERATOR,
+                            true
+                    );
+            newUser = userRepository.save(newUser);
+            newModerator = userRepository.save(newModerator);
+            newUniversity.setEnrolledUsers(Set.of(newUser,newModerator));
+
+            universityRepository.save(newUniversity);
+            performAs(Role.MODERATOR, Set.of(newUniversity.getId()), newModerator.getId());
+            performDelete(newUser.getId()).andExpect(status().is2xxSuccessful());
+        }
+        @Test
+        void delete_UniversityAdministrator_IsForbidden() throws Exception {
+            performAs(Role.MODERATOR, userId);
+            performDelete(userId).andExpect(status().isForbidden());
+        }
+        @Test
+        void delete_Administrator_Ok() throws Exception {
+            var newUser =
+                    new User(
+                            null,
+                            null,
+                            null,
+                            null,
+                            "TEST_USER"+Instant.now().toString(),
+                            "TEST_USER",
+                            null,
+                            null,
+                            null,
+                            null,
+                            null,
+                            null,
+                            Role.USER,
+                            false
+                    );
+            newUser = userRepository.save(newUser);
+            performAs(Role.ADMIN, userId);
+            performDelete(newUser.getId()).andExpect(status().is2xxSuccessful());
+        }
+
+        @Test
+        void delete_Administrator_Error() throws Exception {
+            var newUser =
+                    new User(
+                           null,
+                            null,
+                            null,
+                            null,
+                            "TEST_USER"+Instant.now().toString(),
+                            "TEST_USER"+Instant.now().toString(),
+                            null,
+                            null,
+                            null,
+                            null,
+                            null,
+                            null,
+                            Role.USER,
+                            true
+                    );
+            newUser = userRepository.save(newUser);
+            performAs(Role.ADMIN, userId);
+            performDelete(newUser.getId()).andExpect(status().is4xxClientError());
+        }
+    }
+    @Nested
+    class ViewUsersListTestClass {
+        //tu coś jest źle, każdy widzi wszystko
```

Comment:
Jeśli dobrze sprawdziłem (unit testem + manualnie), to z poziomu każdego użytkownika mamy dostęp do pełnej listy użytkowników w systemie. Wystarczy przejść na /admin/users

Reply:
Ale jak rozumiem nie jest to oczekiwane zachowanie?

Reply:
Możemy stwierdzić, że oczekiwane, aczkolwiek po UI nie idzie się dostać w to miejsce. Sama część linku "/admin" jest dość kontrowersyjna. No i jeszcze pozostaje fakt, że w ten sposób ktoś może sprawdzić dane innego użytkownika, które przechowujemy więc adresy email, nr telefonów. Wiec klasyfikuje to jako nieoczekiwane zachowanie

Reply:
Wydaje mi się, że zachowanie powinno być takie:
1. Użytkownik anonimowy (niezalogowany) - Forbidden/NotFound
2. Użytkownik uniwersytetu - może pobrać listę, ale podaje mu tylko użytkowników z jego uniwersytetu (potrzebne do contact request handlerów)
3. Administrator uniwersytetu - to samo co wyżej
4. Administrator - może wszystkich pobrać

Więc jeśli tak nie jest, to możesz poprawić. Idealnie chyba by było dodać zmienną konfiguracją w `application,yml`, która informuje, czy użytkownik/moderator uniwersytetu może pobierać użytkowników spoza jego uniwersytetu - jestem w stanie sobie wyobrazić use case, że chcieliby móc tak robić, ale nie jesteśmy w stanie teraz tego stwierdzić.
Ewentualnie trzeba napisać maila do Promotora. Jeśli zdecydowałbyś się pisać, to można od razu zapytać, czy mamy jakieś informacje o serwerze politechniki, na który mielibyśmy zrobić deploy.

Reply:
Po przemyśleniu sprawy myślę, że można zostawić jak jest na ten moment i dopytać promotora, bo dodanie takich ograniczeń na start sprawi, że np. nie będzie można przypisać contact request handlera z innego uniwersytetu (a chyba w tym cały sens tych handlerów i tego, że jest to aplikacja dla konsorcjum).